### PR TITLE
Feature/#140202783/static page management

### PR DESCRIPTION
**Pull request by tech-lady-zz**
#### What does this PR do?
This PR adds static page feature to the reaction app

#### Description of Task to be completed?
Create and edit static pages, such as Contact, About, or Support.

#### How should this be manually tested?
An Admin can create, edit or delete any static page

#### Screenshots (if appropriate)

screen shot 2017-03-22 at 11 52 50 am
screen shot 2017-03-22 at 11 53 25 am
screen shot 2017-03-22 at 11 53 38 am
screen shot 2017-03-22 at 11 54 56 am
screen shot 2017-03-22 at 11 55 09 am

### CODE REVIEW

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  $(".editor-toolbar").append(
+    $("<a/>")
+      .attr("id", "static-pages-submit")
+      .attr("title", "Create")
+      .attr("tabindex", "-1")
+      .addClass("fa fa-floppy-o static-pages-new")
+  );
+});
+Template.staticPages.onDestroyed(()=>{
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: We should have space in-between `=>`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  baseUrl() {
+    return window.location.host;
+  },
+  userPages() {
+    return StaticPages.find({shopId: Reaction.shopId}).fetch();
+  }
+});
+
+Template.staticPages.events({
+  "click .static-pages-new": ()=> {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Space in-between `=>`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    const createdAt = new Date();
+    Meteor.call("insertPage", title, slug, content, shopId, pageOwner,  createdAt, function (err) {
+      if (err) {
+        Alerts.toast(err.message, "error");
+      } else {
+        Alerts.toast("Created New Static Page", "success");
+      }
+    });
+  },
+  "click .static-pages-update": ()=> {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Space in-between `=>`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const _id = $("#static-page-id").val();
+    const title = $("#static-page-title").val();
+    const slug = $("#static-page-slug").val();
+    const shopId = Reaction.shopId;
+    const content = simplemde.value();
+
+    Meteor.call("updatePage", _id, title, slug, content, shopId,  function (err) {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Can we not be consistent with the use of arrow function here?

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      if (err) {
+        Alerts.toast(err.message, "error");
+      } else {
+        Alerts.toast(`Updated ${title}`, "success");
+      }
+    });
+  },
+  "click #static-pages-menu-toggle": (event)=> {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Space in-between `=>`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    event.preventDefault();
+    $("#static-pages-wrapper").toggleClass("active");
+    $("#static-pages-menu-toggle > i").toggleClass("fa-toggle-on fa-toggle-off");
+    $("#static-pages-menu-toggle").toggleClass("static-pages-menu-adjust");
+  },
+  "click .editPage": (event)=> {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Space in-between `=>`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+    const pageDetails = StaticPages.find({_id}).fetch();
+    $("#static-page-title").val(pageDetails[0].title);
+    $("#static-page-slug").val(pageDetails[0].slug);
+    $("#static-page-id").val(pageDetails[0]._id);
+    simplemde.value(pageDetails[0].content);
+    $("#static-pages-submit").removeClass("static-pages-new")
+    .addClass("static-pages-update")
+    .prop("title", "Update");
+  },
+  "click .deletePage": (event)=> {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Space in-between `=>`

File: server/methods/core/staticPage.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
+import { Meteor } from "meteor/meteor";
+import { check } from "meteor/check";
+import {StaticPages } from "/lib/collections";
```
Comment (rowlandekemezie) [inline server/methods/core/staticPage.js]: Space needed here `{ StaticPages }`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+      cancelButtonText: "No",
+      confirmButtonText: "Yes"
+    }, (confirmed) => {
+      if (confirmed) {
+        const _id = $(event.currentTarget).parents("tr").attr("id");
+        Meteor.call("deletePage", _id);
+      }
+    });
+  },
+  "click .addPage": ()=> {
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: Space in-between `=>`

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.html [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  <div id="page-content-wrapper">
+    <div class="page-content">
+      <div class="container-fluid">
+        <div class="row">
+          <div class="col-md-10 col-md-offset-1">
+            <div class="static-pages-input-wrapper" >
+              <div class="form-group">
+                <input class="form-control" id="static-page-title" type="text" placeholder="Title">
+              </div>
+              <input style="display:none" type="text" class="form-control" id="static-page-id">
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/client/templates/staticPageStart.html]: Can we avoid using inline styling.

File: client/modules/accounts/templates/dropdown/dropdown.html [hunk 11]
```diff
@@ -78,3 +78,24 @@
...
+<template name="staticPagesNav">
+  <div class="dropdown" role="menu">
+    <div class="dropdown-toggle" data-toggle="dropdown" data-hover="dropdown" data-delay="1000">
+      <span id="logged-in-display-name"> Pages <b class="caret"></b></span>
+    </div>
+    <div class="dropdown-menu static-pages-dropdown" role="menu">
+      {{#each staticPage in staticPages}}
+      <li class="dropdown-apps-icon">
+        <a href="/pages/{{staticPage.slug}}" id="{{staticPage.title}}" title="{{staticPage.title}}">
+        <i class=""></i>
```
Comment (rowlandekemezie) [inline client/modules/accounts/templates/dropdown/dropdown.html:86]: Are we using this for anything?

File: imports/plugins/included/default-theme/client/styles/staticPages/staticPages.less [hunk 12]
```diff
@@ -0,0 +1,69 @@
+#static-pages-wrapper {
+  padding-left: 250px;
+  transition: all 0.4s ease 0s;
+  overflow-x: scroll;
+  height: 75vh;
+}
+
+#sidebar-wrapper {
```
Comment (rowlandekemezie) [inline imports/plugins/included/default-theme/client/styles/staticPages/staticPages.less:8]: **Nitpick**
I would recommend not using too many negative margins in your application. Flexbox model can come in handy for you

File: imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js (hunk 1 above)
Comment (rowlandekemezie) [reply imports/plugins/included/staticPagesStart/client/templates/staticPageStart.js]: If arrow function works here, I doubt it won't work in other templates

File: imports/plugins/included/staticPagesStart/server/policy.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import { BrowserPolicy } from "meteor/browser-policy-common";
+/*
+ * set browser policies
+ */
+BrowserPolicy.content.allowOriginForAll("*.facebook.com");
```
Comment (rowlandekemezie) [inline imports/plugins/included/staticPagesStart/server/policy.js]: Any reason why we're using these policies? I think we already have these dependencies in our project.

**Review by rowlandekemezie — CHANGES_REQUESTED** [review]
Take note of the comments and respond accordingly

File: client/modules/accounts/templates/dropdown/dropdown.html (hunk 11 above)
Comment (tech-lady-zz) [reply client/modules/accounts/templates/dropdown/dropdown.html:86]: Yes @mentrie, this list the number of pages in the dropdown menu on the navigation bar. It increases as more pages are being created.

**Review by rowlandekemezie — APPROVED** [review]
LGTM

**Pull request merged**